src/ipa/rpi/controller/rpi/sync.cpp

```diff
@@ -119,100 +139,175 @@ void Sync::initialise()
 	socket_ = -1;
 }
 
+/* No idea what this does*/
 void Sync::switchMode([[maybe_unused]] CameraMode const &cameraMode, [[maybe_unused]] Metadata *metadata)
 {
 	syncReady_ = false;
 	frameCount_ = 0;
 	readyCountdown_ = 0;
 }
 
+
+/* Most important part, algorithm*/
 void Sync::process([[maybe_unused]] StatisticsPtr &stats, Metadata *imageMetadata)
 {
 	SyncPayload payload;
-	SyncParams local {};
-	SyncStatus status {};
+	SyncParams local{};
+	SyncStatus status{};
 	imageMetadata->get("sync.params", local);
-
 	if (!frameDuration_) {
 		LOG(RPiSync, Error) << "Sync frame duration not set!";
 		return;
 	}
+	if (mode_ == Mode::Off) {
+		return;
+	}
+
+	if (!initialise) {
```

Comment:
Member variables will all need a trailing underscore (`initialise_`). Might be nice to be clear what is being initialised (there are always so many things being initialised!), so perhaps `socketInitialised_`?